src/components/InputCreator.jsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useState, useRef, useEffect } from "react";
+import validationRules from "../utils/RegEx";
+
+function InputCreator({ type, className, name, outsideRef, inputName }) {
+  const [passwordIcon, setPasswordIcon] = useState("show");
+  const [showImputError, setShowImputError] = useState(false);
+  const passwordEye = useRef();
+  const insideRef = useRef();
+  const [errorMessagePrint, setErrorMessagePrint] = useState();
+
+  useEffect(() => {
+    outsideRef.current.value = undefined;
+    outsideRef.current.validate = validate;
+  });
+
+  function validate() {
+    const inputType = inputName;
+    const inputValue = insideRef.current.value;
+    console.log(`HERE WE GO WITH ${inputType}`);
+    const isRegexOk = function (inputType, inputValue) {
+      const errorMessageFound = [];
+      console.log(`EL ARRAY INICIALIZADO`);
+      console.log(errorMessageFound);
+      console.log("IS REGEXOK RUNNING");
+      if (inputType === "text") {
```

Comment:
@zyroans  - This validation is no longer needed since we are using `inputType` parameter.